crypto/s2n_ecc_evp.c

```diff
@@ -339,7 +344,7 @@ int s2n_ecc_evp_write_params_point(struct s2n_ecc_evp_params *ecc_evp_params, st
 
     DEFER_CLEANUP(EC_KEY *ec_key = EVP_PKEY_get1_EC_KEY(ecc_evp_params->evp_pkey), EC_KEY_free_pointer);
     S2N_ERROR_IF(ec_key == NULL, S2N_ERR_ECDHE_UNSUPPORTED_CURVE);
-
+    GUARD_OSSL(EC_KEY_check_key(ec_key), S2N_ERR_ECDHE_SERIALIZING);
```

Comment:
What if modern EC is supported?  (I'm not sure what that define means in this case).

Reply:
`MODERN_EC_SUPPORTED` refers to version of OpenSSL being atleast 1.1.0. The reason we have this flag is because certain APIs for `EVP_PKEY` was not available in older versions of OpenSSL (specifically for x25519 support). The intention was to keep all `EC_KEY` lower level functions limited to older versions of OpenSSL.  Since there isn't a OpenSSL API for `EVP_PKEY` to validate peer public, I have obtained the `EC_KEY` object from `EVP_PKEY` to validate the key and have moved the cleanup function for `EC_KEY` to all versions, hence requiring the clean-up function to be present for both older and newer versions.